models/experimental/functional_mistral/tt/mistral_helper_funcs.py

```diff
@@ -0,0 +1,88 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import ttnn
+import tt_lib
+import torch
+from models.utility_functions import tt_to_torch_tensor, torch_to_tt_tensor_rm
+
+
+def _reshape_for_broadcast(freqs_cis, x_shape, x_dim):
+    ndim = x_dim
+    assert 1 < ndim
+    assert freqs_cis.shape == (x_shape[1], x_shape[-1]), (
+        freqs_cis.shape,
+        (x_shape[1], x_shape[-1]),
+    )
+    shape = [d if i == 1 or i == ndim - 1 else 1 for i, d in enumerate(x_shape)]
+    return freqs_cis.view(*shape)
+
+
+def get_freqs_cis(freqs_cis, query_shape, key_shape, device, mem_config):
+    freqs_cis = _reshape_for_broadcast(freqs_cis, query_shape, 4)
+
+    freq_real = torch_to_tt_tensor_rm(freqs_cis.real, device)
+    freq_img = torch_to_tt_tensor_rm(freqs_cis.imag, device)
+    freqs_cis = tt_lib.tensor.complex_tensor(freq_real, freq_img)
+
+    freq_real.deallocate()
+    freq_img.deallocate()
+
+    t_one_xq = torch.ones(query_shape)
+    t_one_xq = ttnn.to_device(
+        ttnn.to_layout(ttnn.from_torch(t_one_xq, dtype=ttnn.bfloat16), layout=ttnn.TILE_LAYOUT), device
+    )
+
+    freqs_real = freqs_cis.real
+    freqs_real = tt_to_torch_tensor(freqs_real)
+    freqs_real = ttnn.to_device(
+        ttnn.to_layout(ttnn.from_torch(freqs_real, dtype=ttnn.bfloat16), layout=ttnn.TILE_LAYOUT), device
+    )
```

Comment:
Thiis is not necessary:
```
   freqs_real = freqs_cis.real
     freqs_real = tt_to_torch_tensor(freqs_real)
     freqs_real = ttnn.to_device(
         ttnn.to_layout(ttnn.from_torch(freqs_real, dtype=ttnn.bfloat16), layout=ttnn.TILE_LAYOUT), device
     )
```

`freqs_real` is a `ttl.tensor.Tensor`. Therefore, all that needs to be done is `ttnn.Tensor(freqs_cis.real)`

Please fix all of these conversions everywhere in the code